dsp/src/iir_int.rs

```diff
@@ -1,6 +1,21 @@
 use serde::{Deserialize, Serialize};
 
-pub type IIRState = [i32; 5];
+#[derive(Copy, Clone, Default, Deserialize, Serialize)]
+pub struct IIRState(pub [i32; 5]);
+
+impl IIRState {
+    #[inline(always)]
+    pub fn get_x(&self, index: usize) -> i32 {
+        // x0 is at index 0 in a biquad between updates
+        self.0[index]
+    }
+
+    #[inline(always)]
+    pub fn get_y(&self, index: usize) -> i32 {
+        // y0 is at index 2 in a biquad between updates
+        self.0[2 + index]
```

Comment:
Same here, although I believe that rust would catch us here.

Reply:
In both cases rust will check array bounds (in debug). Using that is the usual style and I don't see how else one would handle the error.

Reply:
More what I'm getting to above is if someone called `get_x(2)`, they'd actually be provided a Y component and no error would be logged. I'm fine with letting rust handle this one, since we can't overreach the bounds here.

Reply:
Right. That should be prevented.
But I noticed I don't need that api anywhere so I'll just remove it.